File: Mobile/pcm_mobile/lib/models/tournament.dart
class Tournament {
  final int id;
  final String name;
  final DateTime startDate;
  final DateTime endDate;
  final int format; // 0: Knockout, 1: RoundRobin
  final double entryFee;
  final double prizePool;
  final int status; // 0=Open, 1=Registering, 2=DrawCompleted, 3=Ongoing, 4=Finished

  Tournament({
    required this.id,
    required this.name,
    required this.startDate,
    required this.endDate,
    required this.format,
    required this.entryFee,
    required this.prizePool,
    required this.status,
  });

  factory Tournament.fromJson(Map<String, dynamic> json) {
    return Tournament(
      id: json['id'],
      name: json['name'],
      startDate: DateTime.parse(json['startDate']),
      endDate: DateTime.parse(json['endDate']),
      format: json['format'],
      entryFee: (json['entryFee'] as num).toDouble(),
      prizePool: (json['prizePool'] as num).toDouble(),
      status: json['status'],
    );
  }
}

File: Backend/PcmBackend/Models/Tournament.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PcmBackend.Models
{
    [Table("358_Tournaments")]
    public class Tournament
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        public TournamentFormat Format { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal EntryFee { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal PrizePool { get; set; }

        public TournamentStatus Status { get; set; } = TournamentStatus.Open;

        public string? Settings { get; set; } // JSON string
    }

    [Table("358_TournamentParticipants")]
    public class TournamentParticipant
    {
        [Key]
        public int Id { get; set; }

        public int TournamentId { get; set; }
        [ForeignKey("TournamentId")]
        public Tournament? Tournament { get; set; }

        public int MemberId { get; set; }
        [ForeignKey("MemberId")]
        public Member? Member { get; set; }

        public string? TeamName { get; set; }
        public bool PaymentStatus { get; set; } // True if paid
    }
}
